crates/wysiwyg/src/dom/nodes/text_node.rs

```diff
@@ -293,10 +293,31 @@ where
     fn fmt_markdown(
         &self,
         buffer: &mut S,
-        _options: &MarkdownOptions,
+        options: &MarkdownOptions,
         _as_message: bool,
     ) -> Result<(), MarkdownError<S>> {
-        buffer.push(self.data.to_owned());
+        if options.contains(MarkdownOptions::NO_ESCAPE) {
+            buffer.push(self.data());
+            return Ok(());
+        }
+
+        let mut escaped = S::default();
+
+        for c in self.data().chars() {
+            match c {
+                // https://spec.commonmark.org/0.30/#ascii-punctuation-character
+                '!' | '"' | '#' | '$' | '%' | '&' | '\'' | '(' | ')' | '*'
+                | '+' | ',' | '-' | '.' | '/' | ':' | ';' | '<' | '=' | '>'
+                | '?' | '@' | '[' | '\\' | ']' | '^' | '_' | '`' | '{'
+                | '|' | '}' | '~' => {
```

Comment:
Though it's possible to escape all of these characters, I'm not sure it's actually necessary. I couldn't find a good source for which characters would always need to be escaped though.